Add plugin: ChatCBT

clairefro opened the pull request:
Add ChatCBT

# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/clairefro/obsidian-chat-cbt-plugin

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js` **contains main.ts** -> main.js from build is ignored in git
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using. I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[MyPluginSettings](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/main.ts#L17)
Rename this class.

[id: "chatcbt-chat",](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/main.ts#L86), [id: "chatcbt-summarize",](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/main.ts#L94)
Obsidian will automatically prefix the command id with the plugin id,
you don't need to include it yourself.

[import Cryptr from "cryptr";](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/util/crypt.ts#L1)
This import is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[await axios.post(url, data, headers);](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/util/chatcbt.ts#L67), [const response: { data: { response: string } }  = await axios.post(url, data);](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/util/chatcbt.ts#L87)
Use the `requestUrl` function from the Obsidian API instead, it will handle some things like CORS automatically.

clairefro commented:
Thanks for the detailed audit

I've implemented all changes except the `requestUrl` one. Are there any docuemnted examples of how to use this? 

I found the [types](https://docs.obsidian.md/Reference/TypeScript+API/requestUrl) in documentation and adjusted the request options accordingly. However, I'm unable to debug why I'm getting 401 errors using the same parameter values I used for the axios call that works fine. With `requestUrl`, I can't see the network requests for some reason to debug

<img width="1011" alt="image" src="https://github.com/obsidianmd/obsidian-releases/assets/9841162/0fedecf8-6231-4b32-b7c0-892c337d0680">

clairefro commented:
@joethei Thanks for the audit. Ready for re-review

- [x] [MyPluginSettings](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/main.ts#L17)
> Rename this class.
--> Changed to `ChatCbtPluginSettings`

- [x] [id: "chatcbt-chat",](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/main.ts#L86), [id: "chatcbt-summarize",](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/main.ts#L94)
> Obsidian will automatically prefix the command id with the plugin id,
you don't need to include it yourself.
--> Removed redundant `chatcbt` prefix 

- [x] [import Cryptr from "cryptr";](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/util/crypt.ts#L1)
> This import is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have isDesktopOnly marked as false in your manifest.json file.
--> updated manifest to `isDesktopOnly`: `true`

- [x] [await axios.post(url, data, headers);](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/util/chatcbt.ts#L67), [const response: { data: { response: string } } = await axios.post(url, data);](https://github.com/clairefro/obsidian-chat-cbt-plugin/blob/4842638125403ac43ef82982abbf8665870c4ae3/src/util/chatcbt.ts#L87)
> Use the requestUrl function from the Obsidian API instead, it will handle some things like CORS automatically.
--> Replaced axios with `requestUrl`

clairefro commented:
Not sure how to add the "Changes made" or "Additional review required" label. 

Hope this one's an easy check so I can share it before OpenAI disappears 😛

joethei commented:
Please have patience, we can't give any ETAs. 
And you don't need to fix the merge conflicts, we will do that before merging.
